python/populate_datatype_fields.py

```diff
@@ -0,0 +1,211 @@
+
+    Args:
+        experiment (Any): existing XNAT experiment
+        xnat_hdr (dict): dict containing all the header info to populate in the data type mrd
+        scan_id (str): custom str e.g. cart_cine_scan
+        mrd_file (str): str for filename of mrd_file containing MR raw data
+    """
+    # Check if scan already exists, otherwise create it with all header data
+    if scan_id in experiment.scans:
+        logger.error(f"XNAT scan {scan_id} already exists")
+        raise NameError(f"XNAT scan {scan_id} already exists")
+
+    # Create the scan with all MRD header data at once
+    logger.info(f"Creating MRD scan {scan_id} with header data")
+
+    # Use the xnat library's proper method for creating scans with data
+    session = experiment.xnat_session
+    scan_uri = f"{experiment.uri}/scans/{scan_id}"
+
+    # Create the scan using PUT request with all header data as query parameters
+    response = session.put(scan_uri, query=xnat_hdr)
+
+    if response.ok:
+        logger.info(f"Successfully created MRD scan: {scan_id}")
+        # Refresh the experiment to see the new scan
+        experiment.clearcache()
+
+        # Get the created scan object
+        scan = experiment.scans[scan_id]
+
+    else:
+        logger.error(f"Failed to create scan: {response.status_code} - {response.text}")
+        raise Exception(f"Failed to create MRD scan: {response.status_code}")
+
+    logger.info(f"Configured MRD scan: {scan_id}")
+
+    # Create resource for MRD files - create the resource first, then upload
+    scan_resource = scan.create_resource("MR_RAW")
+    scan_resource.upload(mrd_file_path, mrd_file_path.name)
+    logger.info(f"Successfully created scan {scan_id} and uploaded MRD file")
+
+
+def main():
+    xnat_server_address = "http://localhost"
+    user = "admin"
+    password = "admin"
+
+    mrd_file_path = (
+        Path(__file__).parent.parent
+        / "test-data"
+        / "ptb_resolutionphantom_fully_ismrmrd.h5"
+    )
+
+    logger.info(f"MRD file path: {mrd_file_path}")
+
+    if not mrd_file_path.exists():
+        raise FileNotFoundError(f"MRD file not found: {mrd_file_path}")
+
+    experiment_date = "2022-05-04"
+    scan_id = "cart_cine_scan"
+
+    project_name = "mrd"
+
+    # Use context manager for automatic connection cleanup
+    with xnat.connect(xnat_server_address, user=user, password=password) as session:
+        logger.info("Connected to XNAT server")
+
+        xnat_project = verify_project_exists(session, project_name)
+        xnat_subject, time_id = create_unique_subject(session, xnat_project)
+        experiment = add_exam(xnat_subject, time_id, experiment_date)
+        # Load MRD header and convert to XNAT format
+        with ismrmrd.Dataset(mrd_file_path, "dataset", create_if_needed=False) as dset:
+            header = dset.read_xml_header()
+            xnat_hdr = mrd_2_xnat(
+                header, os.path.join(os.path.dirname(__file__), "ismrmrd.xsd")
```

Comment:
```suggestion
                header, Path(__file__).parent / "ismrmrd.xsd")
```
It'd be good to stick to `Path` where possible (as we're using `pathlib` everywhere else). `os` tends to return `str` which can be less reliable across different operating systems.